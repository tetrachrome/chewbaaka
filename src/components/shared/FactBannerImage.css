/**
 * FactBannerImage.css
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Aug 11, 2020
 * Updated  : Aug 11, 2020
 */

.FactBannerSmallDimension {
  width: 640px;
}

.FactBannerMediumDimension {
  width: 720px;
}

.FactBannerLargeDimension {
  width: 960px;
}
