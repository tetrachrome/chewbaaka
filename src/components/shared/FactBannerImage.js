/**
 * FactBannerImage.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Aug 11, 2020
 * Updated  : Aug 11, 2020
 */

import React from 'react'

import './FactBannerImage.css'

export default function FactBannerImage({ src, alt, large, centered }) {
  let classNames = ["FullWidth", "DisplayBlock"];

  if ( centered ) {
    classNames.push("Centered");
  }

  const sizeClass = large ? "FactBannerMediumDimension" : "FactBannerSmallDimension";

  classNames.push(sizeClass);

  return (
    <img
      className={classNames.join(" ")}
      src={src}
      alt={alt}
    />
  );
}
