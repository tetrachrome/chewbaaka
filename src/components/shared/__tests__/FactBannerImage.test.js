/**
 * FactBannerImage.test.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Aug 17, 2020
 * Updated  : Aug 17, 2020
 */

import React from 'react'

import { render } from '@testing-library/react'

import renderer from 'react-test-renderer'

import FactBannerImage from '../FactBannerImage'

const href = "https://google.com/image.jpg";
const src = "Some image";

test('renders FactBannerImage component', () => {
  render(
    <FactBannerImage
      src={src}
      href={href}
      large
      centered
    />
  );
});

test('FactBannerImage component snapshot', () => {
  const tree = renderer
    .create(
      <FactBannerImage
        src={src}
        href={href}
        large
        centered
      />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
