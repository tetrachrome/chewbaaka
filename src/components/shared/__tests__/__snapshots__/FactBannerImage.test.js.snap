// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FactBannerImage component snapshot 1`] = `
<img
  className="FullWidth DisplayBlock VerticalCushionPadding Centered FactBannerMediumDimension"
  src="Some image"
/>
`;
